packages/base-driver/lib/basedriver/commands/log.js

```diff
@@ -13,11 +13,12 @@ export function LogMixin (Base) {
    * @implements {ILogCommands}
    */
   class LogCommands extends Base {
-    /**
-     * XXX: dubious
-     * @type {Record<string,LogType<Driver>>}
-     */
-    supportedLogTypes;
+
+    constructor (...args) {
+      super(...args);
+      /** @type {Record<string, LogType<Driver>>} */
+      this.supportedLogTypes = this.supportedLogTypes ?? {};
```

Comment:
I'm not quite sure about this expression. Do we assign the variable to itself?

Reply:
it's this:

```js
if (this.supportedLogTypes === null || this.supportedLogTypes === undefined) {
  this.supportedLogTypes = {};
}
```

Reply:
I was just not quite sure how "this.supportedLogTypes" gets assigned before the constructor

Reply:
seems like it's just a safety mechanism because we can't rule out that it might have been set in some other "mixin"'s constructor and if it has, we leave it alone?

Reply:
No, it doesn't have anything to do with mixins potentially setting the value.  `this.supportedLogTypes` is assigned before the constructor if it is on the constructor's `prototype`.  In the case of `AndroidDriver`, this is the case.  

What Babel seems to do (and this may be correct behavior) is a class field, e.g.,:

```js
class Foo {
  bar = 'baz'
}
```

actually results in this:

```
class Foo {
  constructor() {
     this.bar = 'foo';
  }
}
```

Now, that's fine if something subclasses `Foo` and then assigns its own value:

```js
class Baz extends Foo
  constructor() {
    super();
    this.bar = 'baz';
  }
}
```

or uses a field itself:

```js
class Baz extends Foo {
  bar = 'baz';
}
```

...but the above code is not the same as this (which is what `AndroidDriver` does, and maybe other drivers as well):

```js
class Baz extends Foo {}
Baz.prototype.bar = 'baz';
```

In the above case, the constructor for `Foo` will overwrite `Baz.prototype.bar` on any newly-instantiated `Baz` object.

What this means is we need to be careful with class fields, because unless a subclass _also_ uses class fields (or assignments in its constructor), we'll run into this problem.

Reply:
actually, the android driver does something _really_ weird, which is assigns to its own prototype in its constructor (why?).  however, changing it to assign to the prototype after the `class` definition didn't fix the issue.

Reply:
Here's a playground link to see for yourself how this happens:

https://www.typescriptlang.org/play?filetype=js#code/MYGwhgzhAEBiD29oG8BQ0PQEZgE7QF5oByAM0WIG5V1MAXeAZTtwEsA7AcwAoBKFWpky4ApnQCuudtDoALVhAB0OXNSEBfVJtShIMAEJgAXtBEAPOiPYATGAiRohKwiRxHiWmrqjQAiuPEzUwsrWzhEAU9-QMUAB1x4BjoAT1iRZTwXYgBHALMqL3h2CHgQdJB4HgADQyMALmgAEmR2EQB3aFq+dSreamAikrLFCuroswbm1o7x7t7KaAB6RegAdQAJAE1oAElGaAAVdb3oKvJ4KoB+S5ogA